staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -1693,3 +1694,23 @@ func TestGetCurrentResourceVersionFromStorage(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, currentPodRV, podRV, "didn't expect to see the pod's RV changed")
 }
+
+func TestWaitUntilWatchCacheFreshAndForceAllEvents(t *testing.T) {
+	backingStorage := &dummyStorage{}
+	cacher, _, err := newTestCacher(backingStorage)
+	if err != nil {
+		t.Fatalf("Couldn't create cacher: %v", err)
+	}
+	defer cacher.Stop()
+
+	workerScheduled := make(chan struct{})
+	go func() {
+		close(workerScheduled)
+		cacher.watchCache.Add(makeTestPodDetails("pod1", 105, "node1", map[string]string{"label": "value1"}))
+	}()
+
+	<-workerScheduled
```

Comment:
I don't undertand why you need to wait for worker being scheduled?
The following `waitUntilWatchCache...` will wait for it anyway, so this doesn't seem to be needed.

Reply:
I just wanted to make sure it won't timeout. Now I think that maybe we should slightly change the test to:
- run `waitUntilWatchCacheFreshAndForceAllEvents` and let it timeout (worker won't add anything to the store)
- schedule a worker (also make sure it is running)
- run `waitUntilWatchCacheFreshAndForceAllEvents` again this time expect success.

Reply:
updated the test, ptal.

Reply:
I would still remove `workerScheduled` channel - starting a gorotuine is super fast and the fact that we get to closing channel doesn't guarantee watchCache.Add will get addded immediately after in heavy cpu-starved scenarios anyway.
So I would remove it - it's unnecessary complication.

Reply:
ok, removed and checked with `stress` binary, seems to be okay.